lib/backend/k8s/resources/resources.go

```diff
@@ -100,3 +102,74 @@ func SetCalicoMetadataFromK8sAnnotations(calicoRes Resource, k8sRes Resource) {
 		}
 	}
 }
+
+// Store Calico Metadata in the in the k8s resource annotations for CRD backed resources.
+// This should store all Metadata except for those stored in Annotations and Labels and
+// store them in annotations.
+func StoreCalicoMetadataInK8sAnnotations(res Resource) error {
+	metadataBytes, err := json.Marshal(res.GetObjectMeta().(*metav1.ObjectMeta))
```

Comment:
You don't need to cast this to a metav1.ObjectMeta do you?

Reply:
Thought I had to cast it to a concrete type instead of just having it marshal from an interface. Just tested it, marshaling from the interface works.  I'll fix that.